cmd/alertmanager/main.go

```diff
@@ -474,6 +477,13 @@ func run() int {
 		go disp.Run()
 		go inhibitor.Run()
 
+		err = tracingManager.ApplyConfig(conf)
+		if err != nil {
+			return fmt.Errorf("failed to apply tracing config: %w", err)
+		}
+
+		go tracingManager.Run()
```

Comment:
This has a goroutine leak because when we reload the configuration, we don't stop the current tracing manager. Instead, we start to accumulate goroutines blocked on `<-m.done`. I was able to see this in `/debug/pprof`.

Reply:
OK so the issue is that when the configuration doesn't change, `ApplyConfig` returns `nil`, but we still start another goroutine for `Run`.